extensions/micrometer/deployment/src/main/java/io/quarkus/micrometer/deployment/export/StatsdRegistryProcessor.java

```diff
@@ -0,0 +1,55 @@
+package io.quarkus.micrometer.deployment.export;
+
+import java.util.function.BooleanSupplier;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.pkg.steps.NativeBuild;
+import io.quarkus.micrometer.deployment.MicrometerRegistryProviderBuildItem;
+import io.quarkus.micrometer.runtime.MicrometerRecorder;
+import io.quarkus.micrometer.runtime.config.MicrometerConfig;
+import io.quarkus.micrometer.runtime.export.StatsdMeterRegistryProvider;
+
+/**
+ * Add support for the StatsD Meter Registry. Note that the registry may not
+ * be available at deployment time for some projects: Avoid direct class
+ * references.
+ */
+public class StatsdRegistryProcessor {
+    private static final Logger log = Logger.getLogger(StatsdRegistryProcessor.class);
+
+    static final String REGISTRY_CLASS_NAME = "io.micrometer.statsd.StatsdMeterRegistry";
+    static final Class<?> REGISTRY_CLASS = MicrometerRecorder.getClassForName(REGISTRY_CLASS_NAME);
+
+    public static class StatsdRegistryEnabled implements BooleanSupplier {
+        MicrometerConfig mConfig;
+
+        @Override
+        public boolean getAsBoolean() {
+            return REGISTRY_CLASS != null && mConfig.checkRegistryEnabledWithDefault(mConfig.export.statsd);
+        }
+    }
+
+    @BuildStep(onlyIf = { NativeBuild.class, StatsdRegistryEnabled.class })
+    MicrometerRegistryProviderBuildItem nativeModeNotSupported() {
+        log.info("The StatsD meter registry does not support running in native mode.");
```

Comment:
What is the reason why we don't support native mode for all these exporters?

Reply:
Because there is work to do in the upstreams to sort out why they don't work (dependencies, transport requirements, low-level protocol choices). Micrometer core and Prometheus work well in native mode, but some of these other (optional) registries have lower level dependencies that we have to work through.

Reply:
We will work through them, but at a slower pace in coordination w/ the upstream(s). Nothing prevents the registries being used in JVM mode in the meanwhile.

Reply:
e.g. https://github.com/quarkusio/quarkus/issues/11923#issuecomment-716404610